### ISSUE TITLE
fix(boundary): `Neumann1D` / `Robin1D` flip the sign of `value` on south and west faces (coordinate-axis gradient instead of the documented outward-normal gradient)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#285]
Part of #275.

## Summary

The docstrings and `docs/boundary_conditions.md` define `value` as `dφ/dn` along the **outward** normal. The ghost cell lies outward of the interior on every face, so `dφ/dn = (ghost − interior)/spacing` on all four faces and the correct fill is `ghost = interior + value·spacing` everywhere. `bc_1d.py` multiplies by `_outward_sign` (−1 on south/west), which flips the sign on those two faces. `value = 0` is unaffected, which is why the elliptic solvers (which only accept `Neumann1D(value=0)`) never noticed.

## Reproduction

```python
import jax.numpy as jnp
from finitevolx import Neumann1D
Ny, Nx, dy, dx = 6, 6, 0.25, 0.25
y = (jnp.arange(Ny) - 1) * dy
phi = jnp.broadcast_to(y[:, None], (Ny, Nx))         # phi = y ; outward normal at the south wall points to -y
out = Neumann1D("south", -1.0)(phi, dx, dy)          # true outward gradient of phi=y at the south wall is -1
print("code ghost row:", float(out[0, 2]), " true phi at ghost:", float(y[0]))
```

```
south, value=-1 (true outward gradient): code ghost = +0.25   true phi at ghost = -0.25
south, value=+1:                          code ghost = -0.25   (reproduces phi=y with the WRONG sign of value)
Robin alpha=0, beta=1, gamma=-1 at south: code ghost = 0.375   true = -0.125   (dy=0.25 grid)
north, value=+1:                          code ghost = 1.625 = true            (north/east are correct)
```

`tests/test_boundary.py:129,139` (`expected = field[1,:] - 4.5`, `field[:,1] - 3.0`) assert the current formula and lock the bug in. `docs/boundary_conditions.md` is internally inconsistent: its derivation `(φ₁ − φ₀)/dn = ±g` gives `φ₀ = φ₁ + g·dn` for south, but the next line says "−" for south/west.

## Root cause

`finitevolx/_src/boundary/bc_1d.py:46` (`_outward_sign`), `:121` (`Neumann1D.__call__`), `:281` (`Robin1D.__call__`).

## Proposed fix

- `Neumann1D`: `ghost = interior + self.value * spacing` on every face; delete `_outward_sign` if nothing else uses it.
- `Robin1D` (`α φ + β dφ/dn = γ`): with `s = β / spacing`, `ghost = (γ − (α/2 − s)·interior) / (α/2 + s)` on every face (no sign flip).
- Flip the two south/west expectations in `tests/test_boundary.py`; add an analytic test (`φ = x + 2y`, all four faces, `value` = the true outward gradient, ghost matches the analytic field to 1e-14).
- Fix the "∓" line in `docs/boundary_conditions.md`.

## Acceptance criteria

- [ ] Analytic outward-gradient test passes on all four faces for Neumann and Robin.
- [ ] Existing `value = 0` behaviour unchanged (elliptic solver tests untouched).
- [ ] CHANGELOG entry (behaviour change for anyone who worked around the sign).
